app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -214,6 +215,41 @@ class FireDatabase : Database() {
         return updateUserData(dataUpdated)
     }
 
+    override fun addUserToFriendsList(userId: String): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        isUserStoredInDatabase(userId).thenAccept {
+            if(!it){
+                future.completeExceptionally(java.lang.Error("The user with $userId is not present on the database."))
+            }else{
+                val newFriend = HashMap<String,Any?>()
+                newFriend.put(userId, null)
+                accessUserAccountFile(userId).child(friendsListFile).updateChildren(newFriend)
+                    .addOnSuccessListener { future.complete(true) }
+                    .addOnFailureListener { future.completeExceptionally(it) }
+            }
+        }
+        return future
+    }
+
+    override fun removeUserToFriendsList(userId: String): CompletableFuture<Boolean> {
```

Comment:
Same remark with the boolean.
Also, I think we should name it `removeUserFromFriendList`.